python/lsst/meas/base/plugins.py

```diff
@@ -72,6 +74,78 @@
                     TransformClass=bl.ApertureFluxTransform, executionOrder=BasePlugin.FLUX_ORDER)
 
 # --- Single-Frame Measurement Plugins ---
+class SingleFrameFPPositionConfig(SingleFramePluginConfig):
+    pass
+
+@register("base_FPPosition")
+class SingleFrameFPPositionPlugin(SingleFramePlugin):
+    '''
+    Algorithm to calculate the position of a centroid on the focal plane
+    '''
+
+    ConfigClass = SingleFrameFPPositionConfig
+
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.SHAPE_ORDER
+
+    def __init__(self, config, name, schema, metadata):
+        SingleFramePlugin.__init__(self, config, name, schema, metadata)
+        self.focalValue = lsst.afw.table.Point2DKey.addFields(schema, name, "Position on the focal plane",
+                "mm")
+        self.focalFlag = schema.addField(name + "_flag", type="Flag", doc="Set to True for any fatal failure")
+        self.detectorFlag = schema.addField(name + "_missingDetector_flag", type="Flag",
+                                            doc="Set to True if detector object is missing")
+
+    def measure(self, measRecord, exposure):
+        det = exposure.getDetector()
+        center = measRecord.getCentroid()
```

Comment:
Following my previous comment, you've switched the order of `det =` and `center =`, which is good. But my point was that `if not det` (the line below) is `True`, you never need to fetch the centroid at all -- you could move the call to `getCentroid()` to after the `else`, for example.